src/posit/connect/oauth/associations.py

```diff
@@ -1,78 +1,106 @@
+from ._types_context_integration import IntegrationContext
 
+if TYPE_CHECKING:
+    from ..context import Context
 
-class IntegrationAssociations(Resources):
+
+class Association(ResourceDict):
+    class _Attrs(TypedDict, total=False):
+        app_guid: str
+        """The unique identifier of the content item."""
+        oauth_integration_guid: str
+        """The unique identifier of an existing OAuth integration."""
+        oauth_integration_name: str
+        """A descriptive name that identifies the OAuth integration."""
+        oauth_integration_description: str
+        """A brief text that describes the OAuth integration."""
+        oauth_integration_template: str
+        """The template used to configure this OAuth integration."""
+        created_time: str
+        """The timestamp (RFC3339) indicating when this association was created."""
+
+    def __init__(self, ctx: Context, /, **kwargs: Unpack["Association._Attrs"]) -> None:
+        super().__init__(ctx, **kwargs)
+
+
+class IntegrationAssociations(ContextP[IntegrationContext]):
     """IntegrationAssociations resource."""
 
-    def __init__(self, params: ResourceParameters, integration_guid: str) -> None:
-        super().__init__(params)
-        self.integration_guid = integration_guid
+    def __init__(self, ctx: IntegrationContext) -> None:
+        super().__init__()
+        self._ctx = ctx
 
-    def find(self) -> List[Association]:
+    def find(self) -> list[Association]:
         """Find OAuth associations.
 
         Returns
         -------
-        List[Association]
+        list[Association]
         """
-        path = f"v1/oauth/integrations/{self.integration_guid}/associations"
-        url = self.params.url + path
+        path = f"v1/oauth/integrations/{self._ctx.integration_guid}/associations"
+        url = self._ctx.url + path
 
-        response = self.params.session.get(url)
+        response = self._ctx.session.get(url)
         return [
             Association(
-                self.params,
+                self._ctx,
                 **result,
             )
             for result in response.json()
         ]
 
 
-class ContentItemAssociations(Resources):
+class ContentItemAssociations(ApiCallMixin, ContextP[ContentItemContext]):
     """ContentItemAssociations resource."""
 
-    def __init__(self, params: ResourceParameters, content_guid: str) -> None:
-        super().__init__(params)
-        self.content_guid = content_guid
+    @classmethod
+    def _api_path(cls, content_guid: str) -> str:
+        return f"v1/content/{content_guid}/oauth/integrations/associations"
 
-    def find(self) -> List[Association]:
+    def __init__(self, ctx: ContentItemContext) -> None:
+        super().__init__()
+        self._ctx = ctx
+        self._path = self._api_path(ctx.content_guid)
+
+    # TODO-barret-future-q: Should this be inherited from ActiveFinderSequence? (It would add find_by)
+    def find(self) -> list[Association]:
```

Comment:
Should this be inherited from ActiveFinderSequence? (It would add find_by) . We could have `.find()` and `.find_by()` be separated out into two `ActiveFinder*` classes.